src/main/java/io/fabric8/maven/docker/WatchMojo.java

```diff
@@ -176,6 +176,9 @@ private Runnable createBuildWatchTask(final ServiceHub hub, final ImageWatcher w
             throws MojoExecutionException {
         final ImageConfiguration imageConfig = watcher.getImageConfiguration();
         final AssemblyFiles files = hub.getArchiveService().getAssemblyFiles(imageConfig, mojoParameters);
+        if (files.isEmpty()) {
+            log.error("No assembly files for " + imageConfig + ". Are you sure you invoked this goal after the `package` goal?");
+        }
```

Comment:
Oops, missed that it was on the WatchMojo (not BuildMojo). So maybe we could simply use a forked lifecycle for watch ? So that we can simply call `docker:watch` instead of `package docker:build docker:watch` ? (package and build would be called implicitely). 

Since `docker:watch` is not supposed to be bound to an execution phase there should be no problem with forking.

wdyt ?